nav2_mppi_controller/include/nav2_mppi_controller/critic_data.hpp

```diff
@@ -47,7 +42,7 @@ struct CriticData
   const models::Trajectories & trajectories;
   const models::Path & path;
 
-  xt::xtensor<float, 1> & costs;
+  Eigen::ArrayXf & costs;
```

Comment:
Did we want to try if using `1f` rather than the `X` for fixed size would help performance?

Reply:
https://eigen.tuxfamily.org/dox/group__arraytypedefs.html 

I think it should be `ArrayXf` for 1xN and `Array2Xf` (or `ArrayXf`) for 2xN

Reply:
Yes, that's true. But in this case, we don't know the length of `costs` that's why I've used `ArrayXf`. Similarly, for `state` and `trajectories` variables, both rows and columns are not known at the compile time, so we have to use `ArrayXXf` only.